.github/workflows/ci.yml

```diff
@@ -3,67 +3,119 @@ on: [push, pull_request]
 name: CI
 
 jobs:
-  check:
-    name: Check
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-        name: "source checkout"
-      - uses: actions-rs/toolchain@v1
-        name: "toolchain installation"
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-      - uses: actions-rs/cargo@v1
-        name: "cargo check"
-        with:
-          command: check
-
   lints:
     name: Lints
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-        name: "source checkout"
-      - uses: actions-rs/toolchain@v1
-        name: "toolchain installation"
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: stable
           override: true
           components: rustfmt, clippy
 
-      - uses: actions-rs/cargo@v1
-        name: "cargo fmt"
+      - name: Run rustfmt
+        uses: actions-rs/cargo@v1
         with:
           command: fmt
           args: --all -- --check
 
-      - uses: actions-rs/cargo@v1
-        name: "cargo clippy"
-        continue-on-error: true  # TODO: Remove once we have fixed the mall
+      - name: Run clippy
+        uses: actions-rs/cargo@v1
+        continue-on-error: true  # TODO: Remove once we have fixed them all
         with:
           command: clippy
 
-  test:
-    name: Tests
+  check:
+    name: Checks
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-        name: "source checkout"
+      - name: Checkout sources
+        uses: actions/checkout@v2
         with:
           submodules: recursive
-      - uses: actions-rs/toolchain@v1
-        name: "toolchain installation"
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: stable
           override: true
-      - uses: actions-rs/cargo@v1
-        name: "cargo test"
+          components: rustfmt, clippy
+
+      - name: Configure Environment
+        id: env
+        run: |
+          echo "CIRCT_SYS_CIRCT_DIR=$GITHUB_WORKSPACE/circt" >> $GITHUB_ENV
+          echo "::set-output name=circt-hash::$(git rev-parse @:./circt)"
+          echo "::set-output name=llvm-hash::$(cd circt && git rev-parse @:./llvm)"
+
+      # Fetch LLVM from the cache if possible, otherwise rebuild it.
+      - name: Fetch LLVM from cache
+        id: cache-llvm
+        uses: martijnhols/actions-cache/restore@v3
+        with:
+          path: circt/llvm/build
```

Comment:
Yeah excellent point. I took this more or less verbatim from the CIRCT project's CI. But since we're just using LLVM and CIRCT as a library here, I totally agree that it makes sense to try and use the installed version.